playjson/src/main/scala/io/finch/playjson/package.scala

```diff
@@ -11,10 +13,11 @@ package object playjson {
    * @tparam A the type of the data to decode into
    */
   implicit def decodePlayJson[A](implicit reads: Reads[A]): Decode.Json[A] =
-    // TODO: Eliminate toString conversion
-    // See https://github.com/finagle/finch/issues/511
-    // PlayJson can parse from byte[] automatically detecting the charset.
-    Decode.json((b, cs) => Try(Json.parse(BufText.extract(b, cs)).as[A]))
+    Decode.json({ (b, cs) => Try(Json.parse(Buf.ByteArray.Shared.extract(b)).as[A])
+      val buf = ChannelBufferBuf.Owned.extract(b)
+      if (buf.hasArray) Try(Json.parse(new String(buf.array(), 0, buf.readableBytes(), cs)).as[A])
+      else Try(Json.parse(buf.toString(cs)).as[A])
+    })
```

Comment:
@clhodapp I played around with it a little. Do you think this version makes the intent more clear?
```scala
val buf = ChannelBufferBuf.Owned.extract(b)
val bufAsString = Try(
  if (buf.hasArray) new String(buf.array(), 0, buf.readableBytes(), cs)
  else buf.toString(cs)
)
bufAsString.map(Json.parse(_).as[A])
```